sockeye/train.py

```diff
@@ -56,6 +58,12 @@ def _build_or_load_vocab(existing_vocab_path: Optional[str], data_path: str, num
         vocabulary = sockeye.vocab.vocab_from_json(existing_vocab_path)
     return vocabulary
 
+def _dict_difference(dict1: Dict, dict2: Dict):
```

Comment:
This does not work because we want to compare the values. In fact the keys normally won't differ as they will be set with default values. I put the two ways mainly for "new versions", when arguments have been added or removed.